File: src/domain/repositoryInterfaces/IEmployeeRepository.ts
import { Employee, NewEmployee, UpdatedEmployee } from "../models/Employee";

interface IEmployeeRepository {
  findEmployeeForUpdateById(id: number) : Promise<UpdatedEmployee | null>;
  updateEmployee(id: number, employee: UpdatedEmployee): Promise<Employee>;
  createEmployee(employee: NewEmployee): Promise<Employee>;
  deleteEmployee(id: number): Promise<Employee | undefined>;
}

export default IEmployeeRepository;

File: src/wsample.domain/Infrastructure/IEmployeeRepository.cs
using wsample.domain.Models;

namespace wsample.domain.Infrastructure
{
    public interface IEmployeeRepository
    {
        public Task<Employee?> FindEmployeeByIdAsync(int id);
        public Task UpdateEmployeeAsync(Employee updateEmployee);
        public Task<int> CreateEmployeeAsync(Employee newEmployee);
        public Task DeleteEmployeeAsync(int id);
    }
}
